auth/src/com/sixsq/slipstream/auth/internal.clj

```diff
@@ -10,13 +10,20 @@
   (:import (java.util Date TimeZone)
            (java.text SimpleDateFormat)))
 
+(defn- xor
+  [a b]
+  (and (not (and a b))
+       (or a b)))
+
 (defn- extract-credentials
   [request]
-  (uh/select-in-params request [:user-name :password]))
+  (let [username (->> request :params ((juxt :user-name :username)) (apply xor))]
```

Comment:
Congrats for that one. 👏😳

Do we need to support the legacy `user-name` param?
If yes, do we really want to cancel the value if both are present? Is it really worth the complexity?
If yes, is there a test for a request with both params `username` and `user-name`?

If we only want to give preference to the new param `username` with the legacy `user-name` as a fallback, following code would be probably simpler:

```clojure
(->> request :params ((some-fn :username :user-name)))
```